cmd/service/metro/metro.go

```diff
@@ -104,6 +105,7 @@ func Run(ctx context.Context) {
 	go func() {
 		sig := <-sigCh
 		logger.Ctx(ctx).Infow("received a signal, stopping metro", "signal", sig)
+		component.GracefulShutdown(fmt.Errorf("OS Signal error"))
```

Comment:
Why not send a signal to the error channel from here? Is the function intended to achieve any non-blocking flows?

Reply:
I created a new method, in case we want to add something else to gracefully shutdown a component, then we can add that in the same method, rather than having all those lines added in this go routine.